modules/core/ShaderSSLT.cpp

```diff
@@ -0,0 +1,21 @@
+#include "ShaderSSLT.h"
+#include "macroses.h"
+
+namespace rt
+{
+	Vec3f CShaderSSLT::shade(const Ray& ray) const
+	{
+		Vec3f res = Vec3f::all(0);
+
+		Vec3f color = m_color;
+
+		Ray I(ray.hitPoint(), ray.dir, ray.counter + 1);
+		res = color + reTrace(I);
```

Comment:
This one makes the object to glow: you add light here. The better choice would be to mix the color of the object and the color of the re-traced ray via a linear combination alpha * color + beta * reTrace(I), where alpha + beta = 
For the coefficients we can use the `m_opacity` variable , e.g. as follows:
`res = m_opacity * color + (1.0f - m_opacity) * reTrace(I);`